compaction/compaction_manager.cc

```diff
@@ -1024,16 +1024,22 @@ future<> compaction_manager::perform_sstable_scrub(column_family* cf, sstables::
 }
 
 future<> compaction_manager::remove(column_family* cf) {
+    auto handle = _compaction_state.extract(cf);
```

Comment:
> So that the compaction_state won't be found from this point on, while stopping the ongoing compaction.

Why is that bad?

Reply:
`compaction_manager::remove` permanently removes `table*` from the compaction manager.
It's called when the table is being dropped. So we don't want to start any new compaction task for that table instance.

Reply:
```c++
future<> compaction_manager::remove(column_family* cf) {
    // We need to guarantee that a task being stopped will not retry to compact
    // a column family being removed.
    // The requirement above is provided by stop_ongoing_compactions().
    _postponed.erase(cf);

    // Wait for all functions running under with_compaction_disabled_gate to terminate.
    auto close_gate = _compaction_state[cf].with_compaction_disabled_gate.close();
```

So it closes the table-local gate and waits for compactions to complete.

Reply:
yes